frontend/src/lifetime/auto-save.ts

```diff
@@ -0,0 +1,77 @@
+import { AutoSaveDocument, RemoveAutoSaveDocument } from "@/dispatcher/js-messages";
+import { DocumentsState } from "@/state/documents";
+import { EditorState } from "@/state/wasm-loader";
+
+/* eslint-disable no-console */
+const GRAPHITE_INDEXED_DB_NAME = "graphite-indexed-db";
+const GRAPHITE_INDEXED_DB_VERSION = 1;
+const GRAPHITE_AUTO_SAVE_STORE = "auto-save-documents";
+const GRAPHITE_AUTO_SAVE_ORDER_KEY = "auto-save-documents-order";
+
+const databaseConnection: Promise<IDBDatabase> = new Promise((resolve) => {
+	const dbOpenRequest = indexedDB.open(GRAPHITE_INDEXED_DB_NAME, GRAPHITE_INDEXED_DB_VERSION);
+
+	dbOpenRequest.onupgradeneeded = () => {
+		const db = dbOpenRequest.result;
+		if (!db.objectStoreNames.contains(GRAPHITE_AUTO_SAVE_STORE)) {
+			db.createObjectStore(GRAPHITE_AUTO_SAVE_STORE, { keyPath: "details.id" });
+		}
+	};
+
+	dbOpenRequest.onerror = () => {
+		console.error("Graphite IndexedDb error:", dbOpenRequest.error);
+	};
+
+	dbOpenRequest.onsuccess = () => {
+		resolve(dbOpenRequest.result);
+	};
+});
+
+export function createAutoSave(editor: EditorState, documents: DocumentsState) {
```

Comment:
I would recommend changing the name of `createAutoSave` to `createAutoSaveManager` to maintain consistently with the neighboring files. This will also help reduce confusion.